src/grammar/grammar.ohm

```diff
@@ -143,6 +144,11 @@ Tact {
 
     StatementForEach = foreach "(" id "," id "in" Expression ")" "{" Statement* "}"
 
+    StatementDestruct = let "{" ListOf<DestructItem, ","> ","? "}" "=" Expression (";" | &"}")
```

Comment:
Just to check that I understood correctly. The `S` in the following declaration is just the type of variable `s1`:
```
let S {field1: a, field2: b} = s1;
```
and this will create two variables `a` and `b`, with `a` having `s1.field1` and `b` with `s1.field2`. So, is the `S` really necessary if it could be inferred from the type of `s1`? (But I agree that it is better to be explicit).

Reply:
yes, you understand correctly how variables are bound

> So, is the S really necessary if it could be inferred from the type of s1?

We are just following a general syntactical principle here: pattern-matching should correspond to building something. And all our structs need the constructor to be specified. Rust, for instance, uses the same approach for destructuring assignments